### DISCUSSION TITLE
Improvements since mid-2025?

### DISCUSSION THREAD

**Discussion by alexisdrakopoulos** [mlfoundations/open_clip#1184] [Q&A]
It's been over a year since I've checked out CLIP models, have there been any significant improvements/changes? I am looking to fine-tune / train CLIP models for image to image retrieval related tasks.

**Comment by rwightman** [comment]
@alexisdrakopoulos there've been more changes in the past two months than in the previous 2 years. Summarized at the top of the README.

**Reply by alexisdrakopoulos** [reply]
I didn't mean in the repo but in new CLIP models, it would be really nice to have a dinov2/v3 + CLIP joint model

**Reply by rwightman** [reply]
@alexisdrakopoulos there haven't been many that I'm aware of... hence effort to expand the scope of model arch and training code to make it easier to explore if any groups have access to enough compute.

**Reply by alexisdrakopoulos** [reply]
I would love a good way to combine the image + text + optionally label signals. It annoys me that CLIP training is so finicky and sensitive to data quality and performs a lot worse than dinov3 base vs siglip2 fine tuned. I'm quite GPU starved

**Reply by rwightman** [reply]
@alexisdrakopoulos the core of the training focus here I'd consider pre-training and it's all compute & data intensive. The pretrained weights are usually pretty good for downstream fine-tuning, though further contrastive fine-tuning often falls into same pitfalls as pretraining if you don't have enough or good enough data. There are a few new knobs re learning rate-decay that might help contrastive fine-tune work around some forgetting issues...

The newer task arch should make it a bit easier to bolt on an additional label loss to the model train step & data pipe... but not sure that'd yield significant improvements.

**Reply by rwightman** [reply]
The task additions are pretty low-abstraction, coding agents make sense of them quickly. If you work with codex or claude you can quickly get an adapted task that would include a label loss, and it'd be able to plumb it through the wds data pipeline effectively as well to try such an idea out.

**Reply by rwightman** [reply]
multi-label / bag-of-words would probably work as a side loss or added to the text component too...
